swift/storage.py

```diff
@@ -262,13 +262,15 @@ def _save(self, name, content, headers=None):
             content.seek(0)
         else:
             content_type = mimetypes.guess_type(name)[0]
+        content_length = os.fstat(content.fileno()).st_size
```

Comment:
Looking at the Django [docs](https://docs.djangoproject.com/en/1.10/howto/custom-file-storage/#django.core.files.storage._save), that would work.